src/miner_blockhistory.erl

```diff
@@ -0,0 +1,97 @@
+-module(miner_blockhistory).
+
+-behavior(gen_server).
+
+-record(state, {
+    url,
+    historical_block=0,
+    historical_time=0,
+    last_retrieved_time=0
+}).
+
+-define(HISTORICAL_TIME_FETCH_INTERVAL, 3600).
+
+-export([init/1, handle_info/2, handle_cast/2, handle_call/3]).
+
+-export([start_link/0, get_history/0]).
+
+start_link() ->
+    BaseURL = application:get_env(miner, api_base_url, "https://api.helium.io/v1"),
+    URL = string:concat(BaseURL, "/blocks/"),
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [URL], []).
+
+-spec get_history() -> {non_neg_integer(), non_neg_integer()} | undefined.
+get_history() ->
+    gen_server:call(?MODULE, get_history).
+
+%% gen_server callbacks
+
+init([URL]) ->
+    {ok, schedule_check(#state{url=URL}), 15}.
+
+handle_info(check, #state{url=URL}=State) ->
+    case blockchain_worker:blockchain() of
+        undefined ->
```

Comment:
This could probably be handled better by correctly waiting for the chain to be available.

Reply:
its not terrible, the other way is to subscribe to blockchain_events and wait for the `new_chain` event or whatever

Reply:
If you used the add_block event to schedule a check, instead of a 15 second loop (and maybe did a consensus membership check) you could get the exact historical block time you'd need for the next block?